src/main/java/seedu/address/logic/parser/TagCommandParser.java

```diff
@@ -25,32 +31,56 @@ public TagCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_TAG);
 
-        Index index;
-
-        try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, TagCommand.MESSAGE_USAGE), pe);
-        }
-
-        if (!argMultimap.getValue(PREFIX_TAG).isPresent()) {
+        Set<Index> indexSet = new HashSet<>();
+        Set<Tag> tagSet = new HashSet<>();
+        List<String> tagStrings = argMultimap.getAllValues(PREFIX_TAG);
+        if (tagStrings.isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, TagCommand.MESSAGE_USAGE));
         }
 
-        String trimmedTagName = argMultimap.getValue(PREFIX_TAG).get().trim().toLowerCase();
-        boolean isEmpty = trimmedTagName.isEmpty();
-        boolean isTooLong = trimmedTagName.length() > MAX_LENGTH;
+        try {
+            List<String> indexStrings = List.of(argMultimap.getPreamble().trim().split("\\s+"));
+            System.out.println(indexStrings.isEmpty() + indexStrings.toString() + indexStrings.size());
+            if (indexStrings.size() == 1 && indexStrings.get(0).isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, TagCommand.MESSAGE_USAGE));
+            }
+            checkIndexLength(indexStrings);
 
-        if (isEmpty) {
+            for (String indexStr : indexStrings) {
+                if (!indexStr.isEmpty()) {
+                    checkIndex(indexStr);
+                    indexSet.add(ParserUtil.parseIndex(indexStr.trim()));
+                }
+            }
+        } catch (NumberFormatException e) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, TagCommand.MESSAGE_USAGE));
         }
 
-        if (isTooLong) {
-            throw new ParseException(Messages.MESSAGE_INPUT_LENGTH_EXCEEDED);
+        for (String tagString : tagStrings) {
+            tagString = tagString.trim().toLowerCase();
+            boolean isEmpty = tagString.isEmpty();
+            boolean isTooLong = tagString.length() > MAX_LENGTH;
+            if (isEmpty) {
```

Comment:
Can this be directly replaced with tagString.isEmpty() so that we don't need to instantiate another boolean?